astronomer/providers/google/cloud/operators/bigquery.py

```diff
@@ -86,7 +86,14 @@ def execute(self, context: Context) -> None:  # noqa: D102
         hook = _BigQueryHook(gcp_conn_id=self.gcp_conn_id)
 
         self.hook = hook
-        job_id = self._job_id(context)
+        job_id = self.hook.generate_job_id(  # type: ignore[attr-defined]
```

Comment:
Will this work with older version of the google provider?

Reply:
Does the older version of the provider has `generate_job_id` in the Hook?

Reply:
Why did we have to add the type ignore?

Reply:
The older version had a `self._job_id(context)` method that has been moved to hook `generate_job_id`.
type ignore has been added because mypy is unable to get attributes in inherited classes. it's running fine in `make run-mypy` but on circle ci it fails.

Reply:
But with these changes in this PR, if someone used older version of the provider, this will fail, right?

Reply:
>type ignore has been added because mypy is unable to get attributes in inherited classes. it's running fine in make run-mypy but on circle ci it fails.

Are you sure about that?  It is possible that the CI is using the old provider where this method doesn't exist

Reply:
> But with these changes in this PR, if someone used older version of the provider, this will fail, right?

I have updated the google provider version in setup.cfg

Reply:
see now CI failed too :) - https://app.circleci.com/pipelines/github/astronomer/astronomer-providers/2175/workflows/c4283223-3bee-4964-8d92-9882be8790d3/jobs/8259 saying you don't need a typeignore

Reply:
> see now CI failed too :) - https://app.circleci.com/pipelines/github/astronomer/astronomer-providers/2175/workflows/c4283223-3bee-4964-8d92-9882be8790d3/jobs/8259 saying you don't need a typeignore

Understood with change in google provider version mypy passed since `generate_job_id` exists.